ocaml/sdk-gen/java/templates/Types.mustache

```diff
@@ -0,0 +1,253 @@
+{
+   /**
+    * Interface for all Record classes
+    */
+   public interface Record
+   {
+       /**
+        * Convert a Record to a Map
+        */
+       Map<String, Object> toMap();
+   }
+   /**
+    * Base class for all XenAPI Exceptions
+    */
+   public static class XenAPIException extends IOException {
+       public final String shortDescription;
+       public final String[] errorDescription;
+       XenAPIException(String shortDescription)
+       {
+           this.shortDescription = shortDescription;
+           this.errorDescription = null;
+       }
+       XenAPIException(String[] errorDescription)
+       {
+           this.errorDescription = errorDescription;
+           if (errorDescription.length > 0)
+           {
+               shortDescription = errorDescription[0];
+           } else
+           {
+               shortDescription = "";
+           }
+       }
+       public String toString()
+       {
+           if (errorDescription == null)
+           {
+               return shortDescription;
+           } else if (errorDescription.length == 0)
+           {
+               return "";
+           }
+           StringBuilder sb = new StringBuilder();
+           for (int i = 0; i < errorDescription.length - 1; i++)
+           {
+               sb.append(errorDescription[i]);
+           }
+           sb.append(errorDescription[errorDescription.length - 1]);
+           return sb.toString();
+       }
+   }
+
+   /**
+    * Thrown if the response from the server contains an invalid status.
+    */
+   public static class BadServerResponse extends XenAPIException
+   {
+       public BadServerResponse(JsonRpcResponseError responseError)
+       {
+           super(String.valueOf(responseError));
+       }
+   }
+
+   /**
+   * Checks the provided server response was successful. If the call
+   * failed, throws a XenAPIException. If the server
+   * returned an invalid response, throws a BadServerResponse.
+   * Otherwise, returns the server response as passed in.
+   */
+   public static void checkError(JsonRpcResponseError response) throws XenAPIException, BadServerResponse
+   {
+        var errorData = response.data;
+        if(errorData.length == 0){
+            throw new BadServerResponse(response);
+        }
+        var errorName = response.message;
+
+        {{#errors}}
+        if (errorName.equals("{{{name}}}")){
+            {{#err_params}}
+            String p{{{index}}} = errorData.length > {{{index}}} ? errorData[{{{index}}}] : "";
+            {{/err_params}}
+            throw new Types.{{{class_name}}}({{#err_params}}p{{{index}}}{{^last}}, {{/last}}{{/err_params}});
+        }
+        {{/errors}}
+
+        // An unknown error occurred
+        throw new Types.XenAPIException(errorData);
+    }
+
+    {{#enums}}
+    public enum {{{class_name}}} {
+        /**
+         * The value does not belong to this enumeration
+         */
+        @JsonEnumDefaultValue
+        UNRECOGNIZED,
```

Comment:
Here we use "UNRECOGNIZED" as default value for Jave SDK , can we use same default Enum value for Go https://github.com/xapi-project/xen-api/pull/5981? @kc284

Reply:
This is similar to the `unknown_enum_value` added in the Go PR. The result is that there are may enums (e.g. `LatestSyncedUpdatesAppliedState` or `SriovConfigurationMode` etc.) that have both `unknown` and `unrecognized` and I find this confusing from the client's perspective. I think the best way to handle it is like C# where we add `unknown`, but only for the enums that do no already have `unknown`.